collect_app/src/main/java/org/odk/collect/android/tasks/sms/SmsUtils.java

```diff
@@ -0,0 +1,160 @@
+    }
+
+
+    /***
+     * Create the intent that's passed to SMS Manager so that the Delivery SMS
+     * BroadcastReceiver can be triggered.
+     * @param context necessary to create pending intent.
+     * @param instanceId identifies the instance of the form being targeted.
+     * @param messageId identifies the specific messages this intent applies to.
+     * @return PendingIntent
+     */
+
+    public static PendingIntent getDeliveryPendingIntent(Context context, String instanceId, int messageId) {
+        Intent sendIntent = new Intent(SMS_DELIVERY_ACTION);
+        sendIntent.putExtra(SMS_INSTANCE_ID, instanceId);
+        sendIntent.putExtra(SMS_MESSAGE_ID, messageId);
+
+        return PendingIntent.getBroadcast(context, messageId, sendIntent, PendingIntent.FLAG_UPDATE_CURRENT);
+    }
+
+
+    /***
+     * Recreates the sent intent that was passed to SMS Manager to check if it exists.
+     * Usefully in cases such as device restarts.
+     * @param context necessary to create pending intent.
+     * @param instanceId identifies the instance of the form being targeted.
+     * @param messageId identifies the specific messages this intent applies to.
+     * @return true if it exists.
+     */
+    public static boolean checkIfSentIntentExists(Context context, String instanceId, int messageId) {
+        Intent sendIntent = new Intent(SMS_SEND_ACTION);
+        sendIntent.putExtra(SMS_INSTANCE_ID, instanceId);
+        sendIntent.putExtra(SMS_MESSAGE_ID, messageId);
+
+        return PendingIntent.getBroadcast(context, messageId, sendIntent, PendingIntent.FLAG_NO_CREATE) != null;
+    }
+
+    /***
+     * Recreates the delivery intent that was passed to SMS Manager to check if it exists.
+     * Usefully in cases such as device restarts.
+     * @param context necessary to create pending intent.
+     * @param instanceId identifies the instance of the form being targeted.
+     * @param messageId identifies the specific messages this intent applies to.
+     * @return true if it exists.
+     */
+    public static boolean checkIfDeliveryIntentExists(Context context, String instanceId, int messageId) {
+        Intent sendIntent = new Intent(SMS_DELIVERY_ACTION);
+        sendIntent.putExtra(SMS_INSTANCE_ID, instanceId);
+        sendIntent.putExtra(SMS_MESSAGE_ID, messageId);
+
+        return PendingIntent.getBroadcast(context, messageId, sendIntent, PendingIntent.FLAG_NO_CREATE) != null;
+    }
+
+    public static String getDisplaySubtext(SmsEvent event, Context context) {
+        Date date = event.getLastUpdated();
+        SmsProgress progress = event.getProgress();
+
+        if (event.getStatus() == null) {
+            return null;
+        }
+
+        if (event.getProgress().getPercentage() == 100 && event.getStatus().equals(MessageStatus.Sent)) {
+            event.setStatus(MessageStatus.Sending);
+        }
+
+        switch (event.getStatus()) {
```

Comment:
This switch statement should catch `IllegalArgumentException`  we had a similar problem in the past https://github.com/opendatakit/collect/issues/1774 you know strings are added by users using Transifex and some of them might be wrong and cause exceptions so please add it just in case.